pkg/systemkuberes/connector.go

```diff
@@ -0,0 +1,161 @@
+package systemkuberes
+
+import (
+	"context"
+	"fmt"
+
+	dtypes "github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+
+	walrus "github.com/seal-io/walrus/pkg/apis/walrus/v1"
+	walruscore "github.com/seal-io/walrus/pkg/apis/walruscore/v1"
+	"github.com/seal-io/walrus/pkg/clients/clientset"
+	"github.com/seal-io/walrus/pkg/kubeclientset"
+	"github.com/seal-io/walrus/pkg/resourcehandler"
+	"github.com/seal-io/walrus/pkg/system"
+)
+
+func installDefaultKubernetesConnector(
+	ctx context.Context,
+	cli clientset.Interface,
+	project string,
+	envType walruscore.EnvironmentType,
+) (*walrus.Connector, error) {
+	connCli := cli.WalrusV1().Connectors(project)
+
+	config, err := readKubeConfig()
+	if err != nil {
+		return nil, fmt.Errorf("read kube config: %w", err)
+	}
+
+	c := &walrus.Connector{
+		ObjectMeta: meta.ObjectMeta{
+			Namespace: project,
+			Name:      DefaultConnectorName,
+		},
+		Spec: walruscore.ConnectorSpec{
+			Type:                      resourcehandler.ConnectorTypeKubernetes,
+			Description:               "Local Kubernetes",
+			Category:                  walruscore.ConnectorCategoryKubernetes,
+			ApplicableEnvironmentType: envType,
+			ConfigVersion:             "v1",
+			ConfigData: map[string]walruscore.Property{
+				"kubeconfig": {
+					Value: config,
+				},
+			},
+		},
+	}
+
+	conn, err := kubeclientset.Create(
+		ctx,
+		connCli,
+		c,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("install default kubernetes connector: %w", err)
+	}
```

Comment:
under docker run mode, the embedded k3s will change its generated config,
we need to update the local connector config at bootstrap, please use kubeclientset.Update(,,, kubeclientset.WithCreateIfNotExisted[*walrus.Connector]()).